docs/Features.md

```diff
@@ -0,0 +1,216 @@
+Markdown and HTML styles are able to render extracted comments in `<aside>` tags, which can then be
+rendered nicely using CSS.
+
+See the HTML example for an example of one way to render the extracted comments.
+
+## Named entrypoints
+
+By default, the entrypoint of the program is always the unnamed code block. However, this limits the
+output of one input file to always be the same source code. It also means that you can't have a name
+on the entrypoint in the documentation, which could be useful.
+
+To get around this, an entrypoint name can be passed to Yarner on the command line. Then, instead
+of starting at the unnamed code block, it will start at the code block with this name.
+
+Note that if you use a named entrypoint, there is no way to reference the unnamed code blocks as
+macros. You can, however, use the unnamed code blocks to provide examples, for example, to the
+readers of the documentation, so they are still useful.
+
+## Multiple Files, Multiple Entrypoints
+
+By naming code blocks with prefix `file:` followed by a relative path, multiple code files can be created
+from one source file. Each code block with the `file:` prefix is treated as a separate entry point.
+
+~~~md
+```rust - file:src/lib.rs
+fn say_hello() {
+    println!("Hello Literate Programmer!");
+}
+```
+~~~
+
+## File Transclusions
+
+Yarner supports file transclusion to allow for more structured or modular project sources.
+*This feature is currently only supported for Markdown.*
+
+Here, the content of file `src/main.rs.md` would be pulled into the document
+before compilation:
+
+```md
+@{{[src/main.rs](src/main.rs.md)}}
+```
+which would render as a link in the sources file. Or simply:
+```md
+@{{src/main.rs.md}}
+```
+which would not render as a link.
+
+Unnamed entrypoints are renamed to file entrypoints during transclusion.
+It is not required that transcluded files have their own entrypoints.
+
+All code blocks from transcluded files are accessible from the *transcluding* file,
+as well as from other transcluded files.
+
+## Include linked files
+
+Files linked from the main source document are included in the compilation process.
+*This feature is currently only supported for Markdown.*
+
+As an example, Yarner would also compile the file `src/main.rs.md` here:
+
+```md
+* [src/main.rs](src/main.rs.md)
+```
+
+However, files are only included in the compilation if they are referenced by a *relative path*, and the file exists in that location.
+
+## Multiple languages
+
+Some documentation formats allow you to indicate the language that a code block is written in. In
+fact, it is recommended that you always include the language when you write a code block,
+particularly if multiple programming languages are used within the same document.
+
+By properly labelling all code blocks, it is then possible to write a program in multiple
+programming languages at once. Whether this is practical or not remains to be seen, but it is
```

Comment:
Well I certainly wrote multiple programs that contained Rust, C and sometimes C++ code. Often production code is in one language while tooling for testing and deployment is written in another, C++ and Python are common combination. So I think practicality is established even though I am admittedly not sure how much this applies in a scientific context, but then again job submission scripts are Bourne shell scripts no matter what the actual modelling code is written in. (Hopefully not using Bash. I have it on good authority that a certain company from Leipzig uses a web application backend written in Bash and that this is the most probably cause if all of the cities inhabitants will someday be reborn as pieces of butter.)

Reply:
This is from the old docs, but applies to individual code blocks. With file entrypoints, multiple languages are even simpler, and it may be questionable if the original way is still useful.

Different languages would mostly go to different output files, or are there exceptions?

Reply:
Also, the featrue described here is not for "compiling" to multiple languages at the same time, while file entrypoints are.

Reply:
> Different languages would mostly go to different output files, or are there exceptions?

At least, I have never used anything else. Hence I would take up every opportunity for simplification there is and maybe remove the feature entirely or at least tone down the auxiliary discussion here.

Reply:
Sounds reasonable!

Reply:
Just seen that there is already `default_language` in the toml.